pandera/backends/polars/builtin_checks.py

```diff
@@ -180,42 +179,24 @@ def notin(data: PolarsData, forbidden_values: Iterable) -> pl.LazyFrame:
     )
 
 
-@register_builtin_check(
-    error="str_matches('{pattern}')",
-)
-def str_matches(
```

Comment:
I decided to remove the `str_matches` and keep only `str_contains` check with `.str.contains(..., literal=False)` since it works for both strings and patterns

Reply:
I think where we can, check backends should implement the built-in check functions, even if there isn't a direct method supported in the underlying framework.

So the implementation would be something like

```
    if isinstance(pattern, str):
        pattern = f"^{pattern}"
    else:
        pattern = re.compile(f"^{pattern.pattern}")
    return data.dataframe.select(
        pl.col(data.key).str.contains(pattern=pattern)
    )
```